addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -220,29 +262,51 @@ def _get_random_operator(self, lang=None, country_id=None):
         if not self.available_operator_ids:
             return False
 
-        self.env.cr.execute("""SELECT COUNT(DISTINCT c.id), c.livechat_operator_id
+        self.env.cr.execute("""
+            SELECT COUNT(DISTINCT c.id), COUNT(DISTINCT rtc.id) > 0 as in_call, c.livechat_operator_id
             FROM discuss_channel c
             LEFT OUTER JOIN mail_message m ON c.id = m.res_id AND m.model = 'discuss.channel'
+            LEFT OUTER JOIN discuss_channel_member cm ON c.id = cm.channel_id AND cm.partner_id = c.livechat_operator_id
```

Comment:
probably no join on channel, to take all calls into account

Reply:
possibly store partner_id (and guest_id) on rtc to avoid this extra step, and optimize (a lot?) the query as rtc is a small empty table, whereas channel member is big

Reply:
Does not look better (tested with 100 query after 5 warm up queries - 200k channels, 400k members, 20k rtc ):

![image](https://github.com/odoo/odoo/assets/48757558/cc0acc3c-bd9f-4092-b5f2-11ab1e9bc667)